sdk/service/sdk.go

```diff
@@ -24,9 +24,8 @@ func NewSDK(cdc *codec.Codec, client *cosmos.Client) Service {
 }
 
 // Create creates a new service from definition.
-func (s *SDK) Create(req *api.CreateServiceRequest) (*service.Service, error) {
+func (s *SDK) Create(req *api.CreateServiceRequest, accountName, accountPassword string) (*service.Service, error) {
 	// TODO: pass account by parameters
```

Comment:
What do you mean by precheck? It is checked in `BuildAndBroadcastMsg` method. Someone might still call it with wrong name/password and sdk needs to handle that. I think that responsibility for checking that shouldn't be on grpc server-side, as it would require a server to be aware of cosmos client.

Reply:
I'm saying this might be the responsibility of the service SDK (not the grpc API). I get that it totally works like that but for me, it just feels more natural and less "magic" that the SDK first check if you are authorized and then process the user's action. Definitely not something blocking but it would be more explicit and we can have control on the error in that case like "account not found" instead of a weird error about the fact that the account cannot be unlocked (either because of password or address).